scripts/mbedtls_dev/psa_information.py

```diff
@@ -98,21 +98,32 @@ def automatic_dependencies(*expressions: str) -> List[str]:
 
 # Define set of regular expressions and dependencies to optionally append
 # extra dependencies for test case.
+
+# Skip AES test cases which require 192- or 256-bit key
+# if MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH defined
 AES_128BIT_ONLY_DEP_REGEX = r'AES\s(192|256)'
-AES_128BIT_ONLY_DEP = ["!MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH"]
+AES_128BIT_ONLY_DEP = ['!MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH']
+# Skip AES/ARIA/CAMELLIA test cases which require decrypt operation in ECB mode
+# if MBEDTLS_BLOCK_CIPHER_NO_DECRYPT enabled.
+ECB_NO_PADDING_DEP_REGEX = r'(AES|ARIA|CAMELLIA).*ECB_NO_PADDING'
+ECB_NO_PADDING_DEP = ['!MBEDTLS_BLOCK_CIPHER_NO_DECRYPT']
 
 DEPENDENCY_FROM_KEY = {
-    AES_128BIT_ONLY_DEP_REGEX: AES_128BIT_ONLY_DEP
+    AES_128BIT_ONLY_DEP_REGEX: AES_128BIT_ONLY_DEP,
+    ECB_NO_PADDING_DEP_REGEX: ECB_NO_PADDING_DEP,
 }#type: Dict[str, List[str]]
-def generate_key_dependencies(description: str) -> List[str]:
-    """Return additional dependencies based on pairs of REGEX and dependencies.
+def generate_descryption_dependencies(
+        dep_list: List[str],
+        description: str
+    ) -> List[str]:
+    """Return additional dependencies based on test case description and REGEX.
     """
-    deps = []
-    for regex, dep in DEPENDENCY_FROM_KEY.items():
+    for regex, deps in DEPENDENCY_FROM_KEY.items():
```

Comment:
As discussed in chat, this should have been:
```suggestion
    for regex, deps in sorted(DEPENDENCY_FROM_KEY.items()):
```
or something equivalent to have a deterministic order for the dependency list. Otherwise running a generator script twice gives different output.

Alternatively, arguably conceptually simpler:
```
    return sorted(dep_list)
```
below. Although if you're going to sort, I would make `deps` a set instead.
```
    deps = set()
    for regex, deps in DEPENDENCY_FROM_KEY.items():
        if ...: deps.add(dep)
    return sorted(deps)
```